### PR TITLE
Added binary search algorithm to Numerical folder

### PR DESCRIPTION
**Pull request by aish2002**

### CODE REVIEW

File: Numerical/binary_search.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Binary Search Algorithm
+# Input: A sorted array arr and element to be searched,key
```
Comment (rohanbabbar04) [inline Numerical/binary_search.py:2]: A sorted array is not necessary

**Review by rohanbabbar04 — COMMENTED** [review]
Add a sample outpur

File: Numerical/binary_search.py (hunk 1 above)
Comment (sayantani11) [reply Numerical/binary_search.py:2]: > A sorted array is not necessary

A sorted array is necessary for binary search, if the user doesn't provide it, the code should implement sort

File: Numerical/binary_search.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        if arr[mid]==key:
+            return mid
+        elif arr[mid]>key:
+            last =mid-1
+        else :
+            first =mid+1
+        mid= (first+last)//2
+    return -1
+
+arr=list(map(int,input("Enter the array elements in sorted order: ").strip().split()))
```
Comment (sayantani11) [inline Numerical/binary_search.py:23]: Instead of asking user for a sorted array, use sort()

**Review by sayantani11 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**